### PR TITLE
prettier fixes, finished linting, fixed some tsc errors

### PR DESCRIPTION
**Pull request by anniehuo**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- edited setDoc in job.ts to allow adding created job ID 
- added types to objecttomap helper function for jobcard and deleted maptoobject helper b/c was unused
- fixed prettier formatting issues in all files
- added babel.config.js and metro.config.js to eslintignore
- fixed some tsc errors
### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"
- look at changes and make sure i didn't break anything 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @surajr10 @themiachou8

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,22 +1,29 @@
 ## What's new in this PR
+
```
Comment (wangannie) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can you create a file `.prettierignore`, add `.github/`,  and revert the changes to this file? We don't want the spacing changes it makes.

File: src/screens/Drafting/Draft.tsx [hunk 2]
```diff
@@ -122,219 +122,253 @@ function DraftScreen({
...
-          <Text style={{ fontSize: 12, fontFamily: 'DMSans_400Regular' }}>
-            Use the toggle to determine what information you want displayed in
-            your public job posting.
-          </Text>
-        </View>
-        <FormProvider {...methods}>
-          <View style={styles.formTop}>
-            <Text style={[styles.formText, { marginLeft: 0, marginBottom: 4 }]}>
-              Category
+        {/* <ScrollView > */}
```
Comment (wangannie) [inline src/screens/Drafting/Draft.tsx]: Remove comment

**Review by surajr10 — APPROVED** [review]
